Update Intercom Dev Doc for Intercom Source v2 (#435)
Amplitude-Developer-Docs/amplitude-dev-center: docs/data/sources/intercom.md

## Considerations

- You need both an Amplitude and an Intercom account.
- You may select either Intercom users' User ID property or their email property to map to Amplitude users' User ID.

> - Users in Amplitude and Intercom must either have the same user IDs or emails for this integration to work. You can select which property to match on; but, if the property does not match for the same user between Amplitude and Intercom, Amplitude interprets the user as new. 

Reviewer: ```suggestion
- Users in Amplitude and Intercom must either have the same user ID or email for this integration to work. You can choose which property to match on. If the property doesn't match for the same user between Amplitude and Intercom, Amplitude interprets the user as new. 
```
